state-reconstruct-fetcher/src/database.rs

```diff
@@ -0,0 +1,152 @@
+use std::{convert::Into, ops::Deref, path::PathBuf};
+
+use ethers::types::{U256, U64};
+use eyre::{Report, Result};
+use rocksdb::{Options, DB};
+
+const INDEX_TO_KEY_MAP: &str = "index_to_key_map";
+const KEY_TO_INDEX_MAP: &str = "key_to_index_map";
```

Comment:
Well, you've convinced me we do need the key set... :-) I agree it doesn't have to be a map, but bi-directional containers are more conventional, and optimizing this would limit the interface, specifically the return value of `add_key` (which admittedly isn't used ATM, but comes in handy when debugging)... Overall, I'd consider that optimization premature...